mono/metadata/threadpool-io.c

```diff
@@ -651,20 +664,18 @@ mono_threadpool_io_remove_socket (int fd)
 
 	mono_coop_mutex_lock (&threadpool_io->updates_lock);
 
-	if (!io_selector_running) {
-		mono_coop_mutex_unlock (&threadpool_io->updates_lock);
-		return;
-	}
-
-	update = update_get_new ();
-	update->type = UPDATE_REMOVE_SOCKET;
-	update->data.add.fd = fd;
-	mono_memory_barrier (); /* Ensure this is safely published before we wake up the selector */
+	if (io_selector_running) {
 
-	selector_thread_wakeup ();
+		update = update_get_new (UPDATE_REMOVE_SOCKET, __func__);
+		g_assertf (fd >= 0, "%d", fd);
+		update->data.remove_socket.fd = fd;
```

Comment:
Kind of lost in here, but we should take this change no matter what.
`update->data.add.fd = fd` to `update->data.remove_socket.fd = fd`